faust/transport/consumer.py

```diff
@@ -830,12 +830,11 @@ async def _commit_livelock_detector(self) -> None:  # pragma: no cover
         await self.sleep(interval)
         async for sleep_time in self.itertimer(interval, name="livelock"):
             if not self.app.rebalancing:
-                await self.verify_all_partitions_active()
+                await self.app.loop.run_in_executor(None, self.verify_all_partitions_active)
```

Comment:
Does this prevent the commit livelock detector from failing when the main thread is blocked during state rebuilds? Otherwise, what's the purpose of making this run_in_executor on a new thread instead of asynchronously processing the `verify_all_partitions_active`?